src/features/defi/providers/foxy/components/FoxyManager/Deposit/components/Confirm.tsx

```diff
@@ -81,33 +82,34 @@ export const Confirm: React.FC<ConfirmProps> = ({ onNext, accountId }) => {
       return
     try {
       dispatch({ type: FoxyDepositActionType.SET_LOADING, payload: true })
-      const [txid, gasPrice] = await Promise.all([
-        foxyApi.deposit({
-          amountDesired: bnOrZero(state?.deposit.cryptoAmount)
-            .times(bn(10).pow(asset.precision))
-            .decimalPlaces(0),
-          tokenContractAddress: assetReference,
-          userAddress: accountAddress,
-          contractAddress,
-          wallet: walletState.wallet,
-          bip44Params,
-        }),
-        foxyApi.getGasPrice(),
-      ])
+
+      if (!supportsETH(walletState.wallet))
+        throw new Error(`handleDeposit: wallet does not support ethereum`)
+
+      const txid = await foxyApi.deposit({
+        amountDesired: bnOrZero(state?.deposit.cryptoAmount)
+          .times(bn(10).pow(asset.precision))
+          .decimalPlaces(0),
+        tokenContractAddress: assetReference,
+        userAddress: accountAddress,
+        contractAddress,
+        wallet: walletState.wallet,
+        bip44Params,
+      })
       dispatch({ type: FoxyDepositActionType.SET_TXID, payload: txid })
       onNext(DefiStep.Status)
 
       const transactionReceipt = await poll({
         fn: () => foxyApi.getTxReceipt({ txid }),
-        validate: (result: TransactionReceipt) => !isNil(result),
+        validate: (result: ethers.providers.TransactionReceipt) => !isNil(result),
         interval: 15000,
         maxAttempts: 30,
       })
       dispatch({
         type: FoxyDepositActionType.SET_DEPOSIT,
         payload: {
-          txStatus: transactionReceipt.status === true ? 'success' : 'failed',
-          usedGasFeeCryptoBaseUnit: bnOrZero(gasPrice).times(transactionReceipt.gasUsed).toFixed(0),
+          txStatus: transactionReceipt.status ? 'success' : 'failed',
+          usedGasFeeCryptoBaseUnit: transactionReceipt.gasUsed.toString(),
```

Comment:
so if this is supposed to be the crypto price, we will still need `gasPrice`, but `effectiveGasPrice` exists on the receipt which can be used instead of making the extra call. Just making sure we are returning the inteded value

```
{
  "jsonrpc": "2.0",
  "id": 1,
  "result": {
    "blockHash": "0x5c5cf9960dc594a2b60369d83ca9fa60ef4ff599e09dee6c2c9a839e92fe48d2",
    "blockNumber": "0x1054b8b",
    "contractAddress": null,
    "cumulativeGasUsed": "0xbf5997",
    "effectiveGasPrice": "0x95efb3007",
    "from": "0x982c736b124ae91e7516aff2c4979c95e384f77b",
    "gasUsed": "0x36a26"
  }
}
```

Reply:
2497b4b9950ca9eb8ccd5c177b6c8a7756ccfb6e